js/web/lib/wasm/jsep/webgpu/ops/concat.ts

```diff
@@ -132,31 +120,51 @@ const createConcatProgramInfo = (inputs: readonly TensorView[], axis: number): P
     var indices = ${output.offsetToIndices('global_idx')};
 
     let inputIndex = calculateInputIndex(${indicesAxis});
-    if (inputIndex != 0u) {
-      let sizeInConcatAxis = array<u32, ${sizeInConcatAxis.length}u>(${sizeInConcatAxisStr});
-      ${indicesAxis} -= sizeInConcatAxis[inputIndex - 1u];
-    }
+    if (inputIndex < ${inputs.length}u) {
```

Comment:
will this condition be false?

Reply:
@fs-eire  as we discussed yesterday, if the input tensors have shapes [1,0] and [1, 1] and if the axis=1 then the result shape is [2, 1]. If the zero sized tensors are not required to match dims along non-concat axes. Otherwise, this input is not valid. If we agree that the input is valid then then number of elements in the output is more than the number of elements in the input(s) requiring padding. This if condition falls in that scenario. 
My experiments with WASM works 9 out of 10 times with the above expectation except occasional garbage padding output. As you suggested we can require the zero-sized inputs match rank with the reference input. If all the inputs are zero-sized then we required that all inputs are same rank.

Reply:
Rolled back changes made to confirm with WASM, ignoring zero-sized input tensor shape.